Source/Controllers/MainViewControllers/TableRelations/SPTableRelations.m

```diff
@@ -254,14 +267,31 @@ - (IBAction)addRelation:(id)sender
 		[connection setEncoding:@"utf8mb4"];
 	}
 
-	// Get all InnoDB tables in the current database
-    SPMySQLResult *result = [connection queryString:[NSString stringWithFormat:@"SELECT table_name FROM information_schema.tables WHERE table_type = 'BASE TABLE' AND engine = 'InnoDB' AND table_schema = %@ ORDER BY table_name ASC", [[tableDocumentInstance database] tickQuotedString]]];
-    [result setDefaultRowReturnType:SPMySQLResultRowAsArray];
-    [result setReturnDataAsStrings:YES]; // TODO: Workaround for #2699/#2700
-    for (NSArray *eachRow in result) {
-        [refTablePopUpButton addItemWithTitle:[eachRow objectAtIndex:0]];
+    // Get all InnoDB Databases
+    
+    NSArray *theDatabaseList = [connection databases];
+    NSMutableArray *allDatabases;
+    allDatabases = [[NSMutableArray alloc] initWithCapacity:[theDatabaseList count]];
+    for (NSString *databaseName in theDatabaseList)
+    {
+        // If the database is either information_schema or mysql then it is classed as a
+        // system database; similarly, performance_schema in 5.5.3+ and sys in 5.7.7+
+        if (![databaseName isEqualToString:SPMySQLDatabase] &&
+            ![databaseName isEqualToString:SPMySQLInformationSchemaDatabase] &&
+            ![databaseName isEqualToString:SPMySQLPerformanceSchemaDatabase] &&
+            ![databaseName isEqualToString:SPMySQLSysDatabase]) {
```

Comment:
Feels sorta dirty, do we happen to do something similar anywhere else that we could share code with?

Reply:
I 'm agree, it's quite dirty, but I found this condition here : SPDatabaseDocument.m  line 610

Reply:
Is there a reason we need to block a user from creating a foreign key referencing a system table? Does MySQL specifically block creation of such foreign keys?

Reply:
Many reasons : 
- For security reason (But maybe that's a bad good reason) to avoid data corruption
- Foreign Key can only be created for engines that support it. By default Innodb support it, but not MyISAM (for the most famous).
- MariaDB uses the following engine for system tables : 
   - MEMORY
   - Aria
   - Performance_Schema
   All of them don't support foreign key
- MySQL uses the following engine for system tables : 
   - MyISAM
   - CSV
   - Performance_Schema
   - Aria
   All of them don't support foreign key

Reply:
Why would you want to target system database? 😄

Reply:
@gduh That makes sense. Let's share the logic between this and SPDatabaseDocument.m then though, put it in some shared method both can use so that if we have to add one more thing later we only have to add it in one place.

Reply:
After studying the code more deeply, I found something better ;-)